src/MultiFormats.Tests/NetworkProtocolTest.cs

```diff
@@ -0,0 +1,67 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using Google.Protobuf;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace MultiFormats.Tests
+{
+    [TestClass]
+    public class NetworkProtocolTest
+    {
+        [TestMethod]
+        public void Stringing() { Assert.AreEqual("/tcp/8080", new MultiAddress("/tcp/8080").Protocols[0].ToString()); }
+
+        [TestMethod]
+        public void Register_Name_Already_Exists()
+        {
+            ExceptionAssert.Throws<ArgumentException>(() => NetworkProtocol.Register<NameExists>());
+        }
+
+        [TestMethod]
+        public void Register_Code_Already_Exists()
+        {
+            ExceptionAssert.Throws<ArgumentException>(() => NetworkProtocol.Register<CodeExists>());
+        }
+
+        private sealed class NameExists : NetworkProtocol
+        {
+            public override string Name => "tcp";
+            public override uint Code => 0x7FFF;
+            public override void ReadValue(CodedInputStream stream) { }
+            public override void ReadValue(TextReader stream) { }
+            public override void WriteValue(CodedOutputStream stream) { }
+        }
+
+        private sealed class CodeExists : NetworkProtocol
+        {
+            public override string Name => "x-tcp";
+            public override uint Code => 6;
+            public override void ReadValue(CodedInputStream stream) { }
+            public override void ReadValue(TextReader stream) { }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)